### PR TITLE
Add com.wonderlandengine.editor

### PR DESCRIPTION
**Pull request by timothyhale**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.

 Wonderland Engine is a highly performant, accessible and light weight graphics engine for the web. It comes with a desktop editor application and an efficient [WebAssembly](https://webassembly.org/) based runtime.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by timothyhale** [comment]
Hey! I ran the linter, and i am getting errors for not having a developer tag and missing screenshot captions in the   `metainfo.xml`. Both are there and seem correct to me, but i tested on Ubuntu 22.04, maybe i have a too old version of `appstream-cli`?

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: why denyin aarch64?

File: com.wonderlandengine.editor.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    build-options:
+      no-debuginfo: true
+    build-commands:
+      - mkdir ${FLATPAK_DEST}/bin ${FLATPAK_DEST}/share
+      - mv bin/* ${FLATPAK_DEST}/bin
+      - mv lib/* ${FLATPAK_DEST}/lib
+      - mv share/* ${FLATPAK_DEST}/share
+    sources:
+      - type: archive
+        url: https://downloads.wonderlandengine.com/1.2.3/WonderlandEngine-1.2.3-Flatpak.tar.gz
```
Comment (hfiguiere) [inline com.wonderlandengine.editor.yml:44]: can this be built from source?

Comment (timothyhale) [reply com.wonderlandengine.editor.yml:44]: No, the Wonderland Editor is proprietary software.

File: com.wonderlandengine.editor.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        url: https://github.com/glfw/glfw.git
+        tag: 3.3.8
+        commit: 7482de6071d21db77a7236155da44c172a7f6c9e
+
+  - name: editor
+    buildsystem: simple
+    build-options:
+      no-debuginfo: true
+    build-commands:
+      - mkdir ${FLATPAK_DEST}/bin ${FLATPAK_DEST}/share
```
Comment (bbhtt) [inline com.wonderlandengine.editor.yml:38]:
```suggestion
      - mkdir -p ${FLATPAK_DEST}/bin ${FLATPAK_DEST}/share ${FLATPAK_DEST}/lib
```

Comment (bbhtt) [reply com.wonderlandengine.editor.yml:38]: Ah I guess lib is created in glfw module.

**Comment by bbhtt** [comment]
bot, build com.wonderlandengine.editor

**Comment by bbhtt** [comment]
This seems to require login and clicking the create account button seems to do nothing

**Comment by Squareys** [comment]
> This seems to require login and clicking the create account button seems to do nothing

Oh, good point! The "Create Account" button is supposed to be opening the system's default browser, we'll have to use a more canonical way of doing that @timothyhale.

If you click the login button a couple of times, it will switch to offline mode.

**Pull request closed without merging**